tests/utilities/test_cli.py

```diff
@@ -662,9 +663,9 @@ def add_arguments_to_parser(self, parser):
 def test_lightning_cli_optimizers_and_lr_scheduler_with_link_to(tmpdir):
     class MyLightningCLI(LightningCLI):
         def add_arguments_to_parser(self, parser):
-            parser.add_optimizer_args(torch.optim.Adam, nested_key="optim1", link_to="model.optim1")
-            parser.add_optimizer_args((torch.optim.ASGD, torch.optim.SGD), nested_key="optim2", link_to="model.optim2")
-            parser.add_lr_scheduler_args(torch.optim.lr_scheduler.ExponentialLR, link_to="model.scheduler")
+            parser.add_optimizer_args(self.optimizer_registered, nested_key="optim1", link_to="model.optim1")
+            parser.add_optimizer_args(torch.optim.SGD, nested_key="optim2", link_to="model.optim2")
+            parser.add_lr_scheduler_args(self.lr_scheduler_registered, link_to="model.scheduler")
```

Comment:
Can you keep the old calls and just add new ones?

Reply:
Done !